docs_src/design/adr/security/0019-spiffe.md

```diff
@@ -0,0 +1,158 @@
+# Use SPIFFE/SPIRE for On-demand Secret Store Token Generation
+
+## Status
+
+** DRAFT **
+
+## Context
+
+In security-enabled EdgeX,
+there is a component called `security-secretstore-setup`
+that seeds authentication tokens for Hashicorp Vault--EdgeX's secret store--into
+directories reserved for each EdgeX microservice.
+The implementation is provided by a sub-component, `security-file-token-provider`,
+that works off of a static configuration file and an environment variable
+that lists the services that require tokens
+and the policy to be attached to those tokens.
+
+The current solution has some problematic aspects:
+
+* These tokens have an initial TTL of one hour (1h)
+  and become invalid if not used and renewed within that time period.
+  In practical terms,
+  this means that the EdgeX framework must be shut down and restarted completely
+  in order to change the list of running services.
+  It is not possible to delay the start of EdgeX services until a later time,
+  as they will not be able to connect to the EdgeX secret store
+  to obtain required secrets.
+
+* Transmission of the authentication token requires one or more shared file systems
+  between the service and `security-secretstore-setup`.
+  In the Docker implementation,
+  this shared file system is implemented by host-based local storage.
```

Comment:
Maybe clarify and say: ._..is implemented using Docker volume's backed by host-based local storage_?

Reply:
And mention that snaps implement this using content interfaces which are like docker volumes between snaps...

Reply:
Docker volumes and bind mounts to the host are two separate things, but I agree more clarity is needed and I neglected to mention the snap content interface.